AquaNet/src/pages/UserHome.svelte

```diff
@@ -194,218 +196,225 @@
+            width: 50px
+            height: 50px
+            border-radius: 10px
+            object-fit: cover
+
+          // Song info and score
+          > div.info
             flex: 1
-            min-width: 0
+            display: flex
+            justify-content: space-between
             overflow: hidden
-            text-overflow: ellipsis
-            white-space: nowrap
-
-          // Make song score and rank not wrap
-          > div:last-child
-            white-space: nowrap
-
-          @media (max-width: $w-mobile)
-            flex-direction: column
-            gap: 0
-
-            .rank-text
-              text-align: left
 
-        .rank-S
-          // Gold green gradient on text
-          background: linear-gradient(90deg, #ffee94, #ffb798, #ffa3e5, #ebff94)
-          -webkit-background-clip: text
-          color: transparent
-
-        .rank-A
-          color: #ff8a8a
-
-        .rank-B
-          color: #6ba6ff
-
-        .lv
-          min-width: 30px
-          text-align: center
-          background: rgba(var(--lv-color), 0.6)
-          padding: 0 6px
-          border-radius: 10px
-          margin-right: 10px
-
-        span
-          display: inline-block
-          text-align: right
-
-        // Vertical table-like alignment
-        span.rank-text
-          min-width: 40px
-        span.rank-num
-          min-width: 60px
-        span.dx-change
-          min-width: 30px
-
-      span.increased
-        &:before
-          content: "+"
-        color: $c-good
+            // Limit song name to one line
+            > div:first-child
+              flex: 1
+              min-width: 0
+              overflow: hidden
+              text-overflow: ellipsis
+              white-space: nowrap
+
+            // Make song score and rank not wrap
+            > div:last-child
+              white-space: nowrap
+
+            @media (max-width: $w-mobile)
+              flex-direction: column
+              gap: 0
+
+              .rank-text
+                text-align: left
+
+          .rank-S
+            // Gold green gradient on text
+            background: linear-gradient(90deg, #ffee94, #ffb798, #ffa3e5, #ebff94)
+            -webkit-background-clip: text
+            color: transparent
+
+          .rank-A
+            color: #ff8a8a
+
+          .rank-B
+            color: #6ba6ff
+
+          .lv
+            min-width: 30px
+            text-align: center
+            background: rgba(var(--lv-color), 0.6)
+            padding: 0 6px
+            border-radius: 10px
+            margin-right: 10px
+
+          span
+            display: inline-block
+            text-align: right
+
+          // Vertical table-like alignment
+          span.rank-text
+            min-width: 40px
+          span.rank-num
+            min-width: 60px
+          span.dx-change
+            min-width: 30px
+
+        span.increased
+          &:before
+            content: "+"
+          color: $c-good

```

Comment:
Try not use 'content' feature of CSS, it may cause confusion in future maintenance.